sql/parser/datum.go

```diff
@@ -237,7 +237,28 @@ func (d DFloat) IsMin() bool {
 }
 
 func (d DFloat) String() string {
-	return strconv.FormatFloat(float64(d), 'g', -1, 64)
+	s := strconv.FormatFloat(float64(d), 'g', -1, 64)
+	if parsesAsInt(s) {
```

Comment:
Oops sorry It was just a trivial example with 10,000,000: https://play.golang.org/p/REW4jO3OXL

The prec parameter doesn't behave as expected. Superfluous decimal digits are being left off. However, adjusting prec does affect what a 'large exponent' is.
Example:  https://play.golang.org/p/cO6hgKyJRh